### ISSUE TITLE
categoryIds/categories/categories_without_path keys indexing issues

### ISSUE THREAD

**Issue by nishant-lbb** [algolia/algoliasearch-magento-2#627]
**Do you want to request a *feature* or report a *bug*?**
Bug

**Bug: What is the current behavior?**
categoryIds/categories/categories_without_path disappear when products are attached to a specific category via the catalog-> categories ->products in category panel.
These keys are properly updated in algolia when categories are attached via catalog->products pages.

**Bug: What is the expected behavior?**
Category keys should be present irrespective of where the categories are attached to products from.

**Bug: What is the proposed solution?**
I've done some debugging on my own, and observed that getCoreCategories() function in Helper/entity/CategoryHelper.php is returning an array with a single value(Error case) when called via the categories page, while from the products page, it returns an array with 80+ categories(correct behaviour). I'd be happy to provide further details here.

**Feature: What is your use case for such a feature?**

**Feature: What is your proposed configuration entry? The new option to add? What is the behavior?**

**What is the version of Magento and of Algolia extension you are using? Always use the latest version of the extension one before opening a bug issue.**
Magento ver. 2.2.4-dev
Algolia Extension: 1.8.2

**Comment by JanPetr** [comment]
Hello @nishant-lbb,

thank you for the report. Would you be able to test it on the latest version of the extension?
Can you still reproduce the issue there?

Thank you for your help!

**Comment by chaohe** [comment]
@JanPetr I can replicate a similar issue on Magento 2.2.3 Algolia 1.10.0

I observe that the queue entry for reindexing products when they are added/removed in the Category editor...
![image](https://user-images.githubusercontent.com/2146405/55026958-52ec2180-4fc1-11e9-8010-f78779c0489b.png)
...fails to attach full category information (see below, this product is supposed to be in 3 categories)...
![image](https://user-images.githubusercontent.com/2146405/55027775-4d8fd680-4fc3-11e9-8e47-48ff1672524a.png)
This happens with either a single product or multiple products in the queue entry.

However, saving a single product via the Product editor produces two lines in the queue, one with quotes and one without...
![image](https://user-images.githubusercontent.com/2146405/55027508-a874fe00-4fc2-11e9-82dc-0b5454ccebeb.png)
... and this indexes correctly...
![image](https://user-images.githubusercontent.com/2146405/55027610-ebcf6c80-4fc2-11e9-8864-9bdd9407f448.png)

Lastly, attempting to replicate the Product behavior by manually adding a line with quotes to the Category behavior...
![image](https://user-images.githubusercontent.com/2146405/55027830-71531c80-4fc3-11e9-9657-66aa97042585.png)
...also does not seem to work...
![image](https://user-images.githubusercontent.com/2146405/55028065-0a823300-4fc4-11e9-9604-f21c6c4d42e1.png)

To summarize, I can see two differences between the indexing jobs
1. There are usually multiple products when attaching via Category editor, but changing to single does not result in success
2. There are two entries in queue when saving via Product editor, but adding the second entry with quotes to Category editor also does not result in success

**Comment by damcou** [comment]
Hi there , I couldn't reproduce this behaviour and I don't know if you have still the issue as the extension have been updated several times since version 1.8.4. I'll close this ticket but feel free to reopen it if you're still experiencing it.
